backend/app/services/files/s3client.py

```diff
@@ -0,0 +1,189 @@
+                        Bucket=config.S3_BUCKET, Key=file_path)
+                    if response["ResponseMetadata"]["HTTPStatusCode"] == 200:
+                        # Read the content of the S3 object
+                        file_content = await response['Body'].read()
+                        return file_content, response["ContentType"]
+                except Exception as e:
+                    return False, False
+        return False, False
+
+    async def upload_file(self, upload_file: UploadFile):
+        # if mimetype is image upload image
+        if upload_file.content_type in image_mimetypes:
+            return await self._upload_image(upload_file)
+        elif upload_file.content_type in pdf_mimetypes:
+            return await self._upload_with_type(upload_file, "Report")
+        return await self._upload_with_type(upload_file, "Other")
+
+    async def delete_file(self, file_path: str):
+        if file_path.startswith(config.S3_PATH_PREFIX):
+            # delete file_path
+            session = get_session()
+            async with session.create_client(
+                    's3',
+                    region_name=self.region,
+                    endpoint_url=self.s3_endpoint_url,
+                    aws_secret_access_key=self.s3_secret_access_key,
+                    aws_access_key_id=self.s3_access_key_id) as client:
+                response = await client.delete_object(
+                    Bucket=config.S3_BUCKET, Key=file_path)
+                if response["ResponseMetadata"]["HTTPStatusCode"] == 204:
+                    logger.info(
+                        f"File deleted path : {self.s3_endpoint_url}/{config.S3_BUCKET}/{file_path}")
+                    return file_path
+        return False
+
+    #
+    # Private methods
+    #
+
+    async def _convert_image(self, upload_file: UploadFile):
+        request_object_content = await upload_file.read()
+        origin_BytesIo = BytesIO(request_object_content)
+        image = Image.open(origin_BytesIo)
+        data = BytesIO()
+        image.save(data, format="webp", quality=60)
+        return (data, origin_BytesIo)
+
+    async def _get_unique_filename(self,
+                                  filename: str,
+                                  ext: str = "") -> Tuple[str, str]:
+        current_time = datetime.datetime.now()
+        # generate unique name for the file
+        unique_name = str(current_time.timestamp()).replace('.', '')
+        split_file_name = os.path.splitext(filename)
+        ext = ext if ext != "" else split_file_name[1]
+        name = split_file_name[0]
+        uri_component_encoded = split_file_name[0]
+        return (f"{unique_name}{uri_component_encoded}{ext}", name)
+
+    async def _upload_image(self, upload_file: UploadFile):
+        # convert to bytes
+        (data, origin_data) = await self._convert_image(upload_file)
+
+        # Webp converted image
+        if self.with_webp:
```

Comment:
how come the images in the list of experiments are png ?

Reply:
Provided images are PNG files. I just kept the possibility to convert (and store) them in WEBP format if they happen to be too large; this feature is on-hold for now.